File: Longest Consecutive Sequence.cpp
class Solution {
public:
    int helper(unordered_set<int> &h, int x) {
        int cnt = 1;
        int t = x;
        while (h.find(t-1) != h.end()) {
            h.erase(--t);
            cnt++;
        }
        while (h.find(x+1) != h.end()) {
            h.erase(++x);
            cnt++;
        }
        return cnt;
    }

    int longestConsecutive(vector<int> &num) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int ret = 0;
        unordered_set<int> h(num.begin(), num.end());
        
        for (int i = 0; i < num.size(); i++) {
            if (h.find(num[i]) != h.end()) {
                ret = max(ret, helper(h, num[i]));
            }
        }
        return ret;
    }
};

File: Longest Consecutive Sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        if (nums == null || nums.Length == 0) {
            return 0;
        }
        
        var set = new HashSet<int>(nums);
        var result = 1;
        foreach (var num in nums) {
            if (set.Contains(num)) {
                var cnt = 1;
                var x = num - 1;
                while (set.Contains(x)) {
                    set.Remove(x--);
                    cnt++;
                }
                x = num + 1;
                while (set.Contains(x)) {
                    set.Remove(x++);
                    cnt++;
                }
                
                result = Math.Max(result, cnt);
            }
        }
        
        return result;
    }
}
